modules/sdk-coin-ada/src/lib/transactionBuilder.ts

```diff
@@ -131,7 +162,62 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
       vkeyWitnesses.add(CardanoWasm.Vkeywitness.new(vkey, ed255Sig));
     });
     witnessSet.set_vkeys(vkeyWitnesses);
-    this.transaction.transaction = CardanoWasm.Transaction.new(txBody, witnessSet);
+    const txDraft = CardanoWasm.Transaction.new(txBody, witnessSet);
+
+    const linearFee = CardanoWasm.LinearFee.new(
+      CardanoWasm.BigNum.from_str('44'),
+      CardanoWasm.BigNum.from_str('155381')
+    );
+
+    // calculate the fee based off our dummy transaction
+    if (!this._fee) {
+      const fee = CardanoWasm.min_fee(txDraft, linearFee).checked_add(CardanoWasm.BigNum.from_str('6408'));
```

Comment:
txDraft needs to add at least one sigature to get correct size, we need to remove this hardcoded value